polyfill/intersection-observer.js

```diff
@@ -965,16 +965,16 @@ function getParentNode(node) {
     return getFrameElement(node);
   }
 
+  // If the parent has element that is assigned through shadow root slot
+  if (parent && parent.assignedSlot) {
```

Comment:
Also probably this needs to be a `while (parent && parent.assignedSlot)` to handle nested slots.